File: api/Medico.Api/DB/Models/AppUserPermissionGroup.cs
using System;

namespace Medico.Api.DB.Models
{
    public class AppUserPermissionGroup
    {
        public Guid AppUserId { get; set; }

        public Guid PermissionGroupId { get; set; }

        public PermissionGroup PermissionGroup { get; set; }

        public AppUser AppUser { get; set; }

        public bool IsDelete { get; set; }
    }
}

File: client/src/dataModels/appUserPermissionGroup.ts
import { ConvertibleFromEntityModel } from "./baseEntityModel";

export class AppUserPermissionGroup extends ConvertibleFromEntityModel {
    AppUserId: string;
    PermissionGroupId: string;
    IsDelete: boolean;

    constructor(isDelete: boolean = false) {
        super();
        this.AppUserId = ""
        this.PermissionGroupId = "";
        this.IsDelete = isDelete;
    }

    createFromEntityModel(entity: any) {
        const appUserPermissionGroup = new AppUserPermissionGroup(!!entity.isDelete);

        appUserPermissionGroup.PermissionGroupId = entity.PermissionGroupId;
        appUserPermissionGroup.AppUserId = entity.AppUserId;

        return appUserPermissionGroup;
    }
}